spharpy/classes/audio.py

```diff
@@ -0,0 +1,190 @@
+        or real-valued. If ``True`` and `domain` is ``'time'``, the
+        input data will be cast to complex. The default is ``False``.
+
+    References
+    ----------
+    .. [#] F. Zotter, M. Frank, "Ambisonics A Practical 3D Audio Theory
+            for Recording, Studio Production, Sound Reinforcement, and
+            Virtual Reality", (2019), Springer-Verlag
+    .. [#] B. Rafely, "Fundamentals of Spherical Array Processing", (2015),
+            Springer-Verlag
+    .. [#] E.G. Williams, "Fourier Acoustics", (1999), Academic Press
+
+    """
+    def __init__(
+            self,
+            data,
+            sampling_rate,
+            n_max,
+            basis_type,
+            normalization,
+            channel_convention,
+            condon_shortley='auto',
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """
+        Create SphericalHarmonicSignal with data, and sampling rate.
+        """
+        # set n_max
+        if (n_max < 0) or (n_max % 1 != 0):
+            raise ValueError("n_max must be a positive integer")
+        if not data.shape[-2] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few sh coefficients for '
+                             f'{n_max=}. Highest possible n_max is '
+                             f'{int(np.sqrt(data.shape[-2]))-1}')
+        self._n_max = int(n_max)
+
+        # set basis_type
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
+
+        if basis_type not in ["complex", "real"]:
+            raise ValueError("Invalid basis type, only "
+                             "'complex' and 'real' are supported")
+        self._basis_type = basis_type
+
+        # set normalization
+        if normalization not in ["n3d", "maxN", "sn3d"]:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN', but is {normalization}")
+        self._normalization = normalization
+
+        # set channel_convention
+        if channel_convention not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {channel_convention}")
+        self._channel_convention = channel_convention
+
+        # set Condon Shortley
+        if not isinstance(condon_shortley, bool) and condon_shortley != 'auto':
+            raise ValueError(
+                "Condon_shortley has to be a bool, or 'auto'.")
+
+        if condon_shortley == 'auto' and basis_type == 'complex':
+            self._condon_shortley = True
+        elif condon_shortley == 'auto' and basis_type == 'real':
+            self._condon_shortley = False
+
+        Signal.__init__(self, data, sampling_rate=sampling_rate,
```

Comment:
before or after this there should be a check that data has at least 3 dimensions. Otherwise if cant be an SH signal.